### ISSUE TITLE
wrong exit code when linting a mix of existing and non-existing modules

### ISSUE THREAD

**Issue by serkonda7** [PyCQA/pylint#5451]
### Bug description

Pylint ignores the fatal error that one module is not found, if another one is.

### Command used

```shell
poetry run pylint myfile.py moduel_not_exists.py && echo $?
```

### Pylint output

```shell
************* Module foo.py
foo.py:1:0: F0001: No module named foo.py (fatal)

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

0
```

### Expected behavior

The exit code should be 1, as an fatal error occured.

### Pylint version

```shell
pylint 2.12.1
astroid 2.9.0
Python 3.9.8 (main, Nov  7 2021, 15:47:09) 
[GCC 11.2.0]
```

### OS / Environment

Kali (XFCE) with latest VS Code
